### PR TITLE
Add test configurations for deterministic outputs on Dataflow

### PR DESCRIPTION
**Pull request by AnandInguva**
The Beam RunInference Benchmarks provides a dashboard but the tests are not clear in terms of the machine configurations used on Dataflow. 

Add machine type and disable autoscaling to have more deterministic output on dataflow for RunInference Benchmarks.

Also, fix up metric names and units in the RunInference grafana dashboard

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by AnandInguva** [comment]
R: @damccorm 

I was looking at the dashboards and there is some more information missing on the machine type. So I added a  machine type. Also, made autoscaling to None so that we could have more deterministic signals from the test.

I estimated the required number of workers so that the experiments completes in under 30 minutes. The number of workers are also similar to what dataflow scales for these experiments.

File: .test-infra/metrics/grafana/dashboards/perftests_metrics/Python_ML_RunInference_Benchmark_Tests.json [hunk 1]
```diff
@@ -424,7 +424,7 @@
       "steppedLine": false,
       "targets": [
         {
-          "alias": "mean_load_model_latency_milli_seconds_resnet101",
+          "alias": "mean_inference_batch_latency_resnet101",
```
Comment (damccorm) [inline .test-infra/metrics/grafana/dashboards/perftests_metrics/Python_ML_RunInference_Benchmark_Tests.json:427]: I actually think I prefer having the milli_seconds piece - its a valuable piece of information for an observer

**Review by damccorm — COMMENTED** [review]
Just one comment, otherwise LGTM

File: .test-infra/metrics/grafana/dashboards/perftests_metrics/Python_ML_RunInference_Benchmark_Tests.json (hunk 1 above)
Comment (damccorm) [reply .test-infra/metrics/grafana/dashboards/perftests_metrics/Python_ML_RunInference_Benchmark_Tests.json:427]: Oh, that makes sense - thanks for clarifying.

**Comment by damccorm** [comment]
Run Inference Benchmarks

**Comment by AnandInguva** [comment]
Looks like everything works as expected.

**Review by damccorm — APPROVED** [review]

**Pull request merged**